tests/lib/Content/Form/Provider/AbstractGroupedContentFormFieldsProviderTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\ContentForms\Content\Form\Provider;
+
+use Ibexa\Contracts\ContentForms\Data\Content\FieldData;
+use Ibexa\Contracts\Core\Repository\Values\Content\Field;
+use Ibexa\Core\FieldType\TextLine\Value;
+use Ibexa\Core\Helper\FieldsGroups\FieldsGroupsList;
+use Ibexa\Core\Repository\Values\ContentType\FieldDefinition;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Form\FormInterface;
+
+abstract class AbstractGroupedContentFormFieldsProviderTest extends TestCase
+{
+    protected function getFieldsGroupsListMock(): FieldsGroupsList
+    {
+        $mock = $this->createMock(FieldsGroupsList::class);
+        $matcher = $this->exactly(3);
+        $expectedGroups = [1 => 'group_1', 2 => 'group_2', 3 => 'group_2'];
+
+        $mock
+            ->expects($matcher)
+            ->method('getFieldGroup')
+            ->willReturnCallback(static function () use ($matcher, $expectedGroups) {
+                return $expectedGroups[$matcher->getInvocationCount()];
+            });
+
+        return $mock;
+    }
+
+    protected function getFormMockWithFieldData(
+        string $fieldDefIdentifier,
+        string $fieldTypeIdentifier
+    ): FormInterface {
+        $formMock = $this
+            ->getMockBuilder(FormInterface::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $formMock
+            ->expects($this->once())
+            ->method('getViewData')
+            ->willReturn(new FieldData([
+                'field' => new Field(['fieldDefIdentifier' => $fieldDefIdentifier]),
+                'fieldDefinition' => new FieldDefinition(['fieldTypeIdentifier' => $fieldTypeIdentifier]),
+                'value' => new Value('value'),
+            ]));
+
+        $formMock
+            ->expects($this->once())
+            ->method('getName')
+            ->willReturn($fieldDefIdentifier);
+
+        return $formMock;
+    }
+
+    protected function getTestForms(): array
```

Comment:
Nitpick: can be final.

```suggestion
    final protected function getTestForms(): array
```